src/components/PackageSearch.tsx

```diff
@@ -213,7 +213,15 @@ export default function PackageSearch() {
 
           {results.packages.length > 0 ? (
             <>
-              <PackageTable packages={results.packages} />
+              <PackageTable
+                onArchitectureClick={arch => {
+                  handleSelectionChange('arch', arch);
```

Comment:
trigger search callback too

Reply:
Need to think a bit how to make it work. Since both execute at the same time and old state is used while not giving enough time to update the state in the end also.